pygn_handler.py

```diff
@@ -37,10 +37,57 @@
 ### file-scoped configurable variables
 ###
 
+# TODO: breaks pixel configuration
+CALLBACKS = {
+    "1": lambda board: board.fen(),
+    "2": lambda board: chess.svg.board(board=board),
+}
+
 ###
 ### subroutines
 ###
 
+def listen():
+    """
+    Listen for messages on stdin and send response data on stdout.
+    """
+
+    while True:
+        input_str = sys.stdin.readline()
+
+        # Handle terminating characters and garbage.
+        # TODO: make more safe.
```

Comment:
We should probably check for these characters specifically, to catch and handle bad requests properly

Reply:
In the case of `len(input_str) == 0`, this is as documented for `readline()`.  There's not another way to get a zero length.   In the case of newline, we can also check for the character.